### PR TITLE
Fix for bug #44: Symbolic toolbar icons and code refactoring

### PR DESCRIPTION
**Pull request by jakubbrindza**
I have worked on implementing nicer and more representative icons for the current design of GTG.
The perviously used icons were very similar and caused confusion. This should become past with these GTK3 icons.

This applies not only to toolbar icons, but also to the right-click menu icons for corresponding actions.

Previous icons:
gtg-before

Changed into:
gtg-after

Some other new stuff in cancelled tasks:
Before (Done task):
gtg_before_cancel_edit

After (Done task):
gtg_after_cancel_edit

Before (Dismissed task):
gtg_before_cancel_edit_2

After (Dismissed task):
gtg_after_closed_edit_2

### CODE REVIEW

**Comment by izidormatusov** [comment]
Icon for dismissing tasks looks too much similar as icon to delete tasks. Could you experiment if you would have a better result with `Gtk.STOCK_CLOSE` or `Gtk.STOCK_DISCARD`? Can you make the icon when a dimsissed task is selected to be `Gtk.STOCK_REFRESH` as well? It would symbolise putting it back in the work queue. 

There is opportunity to clean GTG from unused icons. Checklist of things to do:

 - [ ] change icons also in GTG/gtk/browser/taskbrowser.ui
   - [ ] gtg-task-done
   - [ ] gtg-task-dismiss
   - [ ] gtg-task-new
 - [ ] remove following icon files:

```
./data/icons/hicolor/16x16/actions/gtg-task-done.png
./data/icons/hicolor/22x22/actions/gtg-task-done.png
./data/icons/hicolor/16x16/actions/gtg-task-undone.png
./data/icons/hicolor/22x22/actions/gtg-task-undone.png
./data/icons/hicolor/16x16/actions/gtg-task-dismiss.png
./data/icons/hicolor/22x22/actions/gtg-task-dismiss.png
./data/icons/hicolor/16x16/actions/gtg-task-undismiss.png
./data/icons/hicolor/22x22/actions/gtg-task-undismiss.png
./data/icons/hicolor/16x16/actions/gtg-task-new.png
./data/icons/hicolor/22x22/actions/gtg-task-new.png

**Comment by jakubbrindza** [comment]
Changes of the DISMISS button into a more distinct button (see the picture). I also deleted the unused icons, this should then solve the bug entirely.

gtg_after_final

**Comment by jakubbrindza** [comment]
Changes to the DISMISS button which is now also more distinct [see the pic] and unified undismiss and undo task buttons to represent the idea of "REFRESHING" them.
I also changed the old gtg-...-... icons in the code for their GTK3 equivalents and removed them from the repo completely, since they are not used and will not be used any more.

gtg_after_final
gtg_after undismiss
gtg-after not done

File: GTG/gtk/browser/browser.py [hunk 1]
```diff
@@ -1305,8 +1305,8 @@ def on_task_cursor_changed(self, selection=None):
         """
         # We unselect all in the closed task view
         # Only if something is selected in the active task list
-        self.donebutton.set_icon_name("gtg-task-done")
-        self.dismissbutton.set_icon_name("gtg-task-dismiss")
+        self.donebutton.set_icon_name(Gtk.STOCK_APPLY)
```
Comment (izidormatusov) [inline GTG/gtk/browser/browser.py]: Above you use set_stock_id for the button, down here you use set_icon_name. Could you use just single way how to set buttons? Is there a difference between those calls?

File: GTG/gtk/browser/taskbrowser.ui [hunk 2]
```diff
@@ -397,7 +397,7 @@
                   <object class="GtkToolButton" id="new_subtask_b">
                     <property name="can_focus">False</property>
                     <property name="label" translatable="yes">New Subtask</property>
-                    <property name="icon_name">gtg-task-new</property>
+                    <property name="icon_name">Gtk.STOCK_NEW</property>
```
Comment (izidormatusov) [inline GTG/gtk/browser/taskbrowser.ui]: I don't think this is going to work, you need to use the literal value here: `gtk-new`

File: GTG/gtk/browser/taskbrowser.ui [hunk 3]
```diff
@@ -809,25 +809,25 @@ Bertrand Rousseau (bertrand.rousseau@gmail.com)</property>
     <property name="visible">True</property>
     <property name="can_focus">False</property>
     <property name="pixel_size">16</property>
-    <property name="icon_name">gtg-task-dismiss</property>
+    <property name="icon_name">Gtk.STOCK_CLOSE</property>
```
Comment (izidormatusov) [inline GTG/gtk/browser/taskbrowser.ui]: same as above

File: GTG/gtk/browser/taskbrowser.ui [hunk 4]
```diff
@@ -809,25 +809,25 @@ Bertrand Rousseau (bertrand.rousseau@gmail.com)</property>
...
   </object>
   <object class="GtkImage" id="bm_img_done">
     <property name="visible">True</property>
     <property name="can_focus">False</property>
     <property name="pixel_size">16</property>
-    <property name="icon_name">gtg-task-done</property>
+    <property name="icon_name">Gtk.STOCK_APPLY</property>
```
Comment (izidormatusov) [inline GTG/gtk/browser/taskbrowser.ui]: same as above

File: GTG/gtk/browser/taskbrowser.ui [hunk 5]
```diff
@@ -809,25 +809,25 @@ Bertrand Rousseau (bertrand.rousseau@gmail.com)</property>
...
   </object>
   <object class="GtkImage" id="bm_img_subtask">
     <property name="visible">True</property>
     <property name="can_focus">False</property>
     <property name="pixel_size">16</property>
-    <property name="icon_name">gtg-task-new</property>
+    <property name="icon_name">Gtk.STOCK_NEW</property>
```
Comment (izidormatusov) [inline GTG/gtk/browser/taskbrowser.ui]: same as above

File: GTG/gtk/browser/taskbrowser.ui [hunk 6]
```diff
@@ -809,25 +809,25 @@ Bertrand Rousseau (bertrand.rousseau@gmail.com)</property>
...
   </object>
   <object class="GtkImage" id="bm_img_task">
     <property name="visible">True</property>
     <property name="can_focus">False</property>
     <property name="pixel_size">16</property>
-    <property name="icon_name">gtg-task-new</property>
+    <property name="icon_name">Gtk.STOCK_NEW</property>
```
Comment (izidormatusov) [inline GTG/gtk/browser/taskbrowser.ui]: same as above

File: GTG/gtk/editor/editor.py [hunk 7]
```diff
@@ -258,24 +258,24 @@ def refresh_editor(self, title=None, refreshtext=False):
         if status == Task.STA_DISMISSED:
             self.donebutton.set_label(GnomeConfig.MARK_DONE)
             self.donebutton.set_tooltip_text(GnomeConfig.MARK_DONE_TOOLTIP)
-            self.donebutton.set_icon_name("gtg-task-done")
+            self.donebutton.set_icon_name(Gtk.STOCK_APPLY)
             self.dismissbutton.set_label(GnomeConfig.MARK_UNDISMISS)
             self.dismissbutton.set_tooltip_text(undismiss_tooltip)
-            self.dismissbutton.set_icon_name("gtg-task-undismiss")
+            self.dismissbutton.set_stock_id(Gtk.STOCK_REFRESH)
```
Comment (izidormatusov) [inline GTG/gtk/editor/editor.py]: Again, why to combine set_icon_name and set_stock_id for the same purpose? Pick single way how to set icons. Consistency makes the code much readable

**Comment by izidormatusov** [comment]
Please, fix the remarks

File: GTG/gtk/browser/browser.py [hunk 8]
```diff
@@ -1242,89 +1242,84 @@ def on_select_tag(self, widget=None, row=None, col=None):
...
-        """Called when selection changes in closed task view.
-
-        Changes the way the selected task is displayed.
+    def on_task_cursor_changed(self, selection):
+        """If there is an active task selected, unselect all the closed tasks
+        and set the status of the selected active task to STA_ACTIVE.
+        If no tasks at all are selected, set status to STA_ACTIVE"""
+
+        if selection.count_selected_rows() > 0 and
+        'closed' in self.vtree_panes:
```
Comment (izidormatusov) [inline GTG/gtk/browser/browser.py]: Can you indent it by 2x4 spaces to make sure that it is visually still part of the condition?

**Comment by izidormatusov** [comment]
Congratulation on your first contribution!

**Pull request closed without merging**